### PR TITLE
Add region role to dashboard alerts

### PR DESCRIPTION
**Pull request by jbalboni**
## Description
Adding a region role with a label improves the accessibility of the page, based on Trevor's research in the linked ticket.

## Testing done
Local testing

## Acceptance criteria
- [x] Alert has a region role and a valid label

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/personalization/dashboard/tests/components/DashboardAlert.unit.spec.jsx [hunk 1]
```diff
@@ -27,13 +27,22 @@ describe('<DashboardAlert />', () => {
     wrapper.unmount();
   });
 
-  it('should pass aXe check', () =>
```
Comment (jbalboni) [inline src/applications/personalization/dashboard/tests/components/DashboardAlert.unit.spec.jsx]: This test started failing with an axe error about the `aria-labelledby` value being invalid. It's definitely not invalid and the axe check passes when using the browser extension on the same html. I tried upgrading axe-core to see if that was the issue, but I still got the bad error. I also didn't see any related GH issues about it. So I'm not sure what's going on, other than I assume it has something to do with JSDom.

Comment (1Copenut) [reply src/applications/personalization/dashboard/tests/components/DashboardAlert.unit.spec.jsx]: I wasn't able to get these tested locally, my API instance kept on giving a warning that I couldn't save apps in progress. I'm out tomorrow but please holler at me Monday if this is still an open item.

File: src/applications/personalization/dashboard/components/DashboardAlert.jsx [hunk 2]
```diff
@@ -26,17 +27,20 @@ const DashboardAlert = ({
   );
 
   const alertContent = content || children;
+  const headerId = `dashboard-alert-header-${id}`;
 
   return (
-    <div className={classes}>
-      <header>
+    <div aria-labelledby={headerId} role="region" className={classes}>
+      <div>
```
Comment (erikphansen) [inline src/applications/personalization/dashboard/components/DashboardAlert.jsx]: is switching this from a `header` to `div` for a11y reasons or other reasons?

**Review by erikphansen — APPROVED** [review]

File: src/applications/personalization/dashboard/components/DashboardAlert.jsx (hunk 2 above)
Comment (jbalboni) [reply src/applications/personalization/dashboard/components/DashboardAlert.jsx]: It was in Trevor's recommendation in the linked issue, I think it's just that it's a confusing landmark when navigating through the page. Not fully sure, though.

Comment (1Copenut) [reply src/applications/personalization/dashboard/components/DashboardAlert.jsx]: @erikphansen && @jbalboni Changing to a div was so assistive device users wouldn't have two headers in their landmarks menu. It was a bit confusing when I was navigating by landmarks, and heard a second header announced inside the main content area.

**Pull request merged**